bandit/formatters/text.py

```diff
@@ -141,9 +141,8 @@ def report(manager, fileobj, sev_level, conf_level, lines=-1):
     """
 
     bits = []
-    issues = manager.get_issue_list(sev_level, conf_level)
 
-    if len(issues) or not manager.quiet:
```

Comment:
By removing this you are changing the behavior such that when someone passes the argument --quiet, the summary is never shown.   The desired behavior for quiet mode is that the summary isn't shown if there are no issues only.  If there are issues found, the summary is always shown.

Reply:
This won't save a ton at runtime, but we should probably be using the `manager.results_count` here instead of the `get_issue_list` method and allocating an additional variable.

Reply:
@ericwb yes, you have a point. It does change the behavior. I wasn't really sure if the new behavior was intended or not.
Anyways, since the behavior was just introduced, it might make sense to discuss it a bit further, just to make sure we get this right.

Ideally we don't have to sacrifice either feature nor performance.

In my humble opinion, changing the program print based on both an invocation parameter and the code that is being inspected does feel a bit unexpected. Me removing that line is this patch is proof for my personal confusion about this behavior.

That being said, I believe you have put some thought into this. So if you confirm, that this is the behavior you intended, I'll dig around a bit and find another solution.

Reply:
@codingjoe A potential interim solution to this would be the following:

```python
if not manager.quiet or manager.results_count(sev_level, conf_level):
    // rest of the logic here
```

It would remove the performance degradation for anyone not using `-q` or `--quiet` at least. Still should come up with a long term solution to solve the overall problem though

Reply:
@tylerwince haha, that would resolve the issue for us, but it would be somewhat strange if `bandit` slows down out of nowhere, because you run it in quiet mode.

Reply:
@codingjoe, tylerwince's suggestion is a good temp fix in my opinion.  At least the performance issue will no longer affect everyone.

Can you update your commit and get complete by today or tomorrow?  If so, I can include in 1.6.1.  Thanks

Reply:
@ericwb and @codingjoe, I have implemented this on your branch @codingjoe. The PR for this can be found [here](https://github.com/codingjoe/bandit/pull/1) and if you just merge that then this PR can get merged. 

Thanks everyone